File: c#/_55_LychrelNumbers.cs
using System;

public class _55_LychrelNumbers{
	public static int Euler55(){
		int res = 0;
		for (int i = 1; i < 10_000; i++){
			if (IsLychrel(i)){
				res++;
			}
		}
		return res;
	}
	
	public static bool IsLychrel(long n){
		for (int i = 0; i < 50; i++){
			n += Reverse(n);
			if (IsPalindrome(n)) return false;
		}
		return true;
	}
	
	public static long Reverse(long n){
		long rev = 0;
		while (n > 0){
			rev = rev * 10 + n % 10;
			n /= 10;
		}
		return rev;
	}
	
	public static bool IsPalindrome(long n){
		return n == Reverse(n);
	}
	
	public static void Main(){
		Console.WriteLine(Euler55());
	}
}

File: java/_55_LychrelNumbers.java
import java.util.*;

public class _55_LychrelNumbers{
    
    public static boolean isPalindrome(long num){
        long rev = 0;
        long n = num;
        while (n > 0){
            long digit = n % 10;
            rev = rev * 10 + digit;
            n = (n - digit) / 10;
        }
        return num == rev;
    }
    
    public static long reverse(long num){
        long rev = 0;
        while (num > 0){
            long digit = num % 10;
            rev = rev * 10 + digit;
            num = (num - digit) / 10;
        }
        return rev;
    }
    
    public static void main(String[] args){
        
        int nums = 0;
        
        for (int i = 1; i < 10000; i++){
            
            long num = i;
            boolean isLychrel = true;
            for (int j = 1; j <= 50; j++){
                long sum = num + reverse(num);
                if (isPalindrome(sum)){
                    // if (j > 20) System.out.println("\t" + i + " " + j + " " + sum);
                    isLychrel = false;
                    break;
                }
                num = sum;
            }
            if (isLychrel){
                // System.out.println("Lychrel: " + i);
                nums++;
            }
        }
        
        System.out.println("Number of integers that do not produce a palindrome within 50 iterations: " + nums); // 249
    }
}